config/tap-driver.sh

```diff
@@ -144,7 +147,7 @@ fi
     else
       exec 2>&3
     fi
-    "$@"
+    run_timeout ${FLUX_TEST_TIMEOUT:-300} "$@"
```

Comment:
SInce this is really just meant for automated CI testing (we don't really see hangs during regular `make check` runs), what if we made the `run_timeout` only active when `FLUX_TEST_TIMEOUT` is set? So there is no change unless it is set, and we set it in travis-ci.

Reply:
I'd be alright with that, like this?

```suggestion
    if test -n "$FLUX_TEST_TIMEOUT" ; then
        run_timeout ${FLUX_TEST_TIMEOUT} "$@"
    else
       "$@"
    fi
```

Reply:
Yeah, exactly what I was thinking. Thanks!! This will be nice